src/Support/ListenerCollectionFactory.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\EventDispatcher\Support;
+
+use Psr\Container\ContainerInterface;
+use ReflectionMethod;
+use Yiisoft\EventDispatcher\Provider\ListenerCollection;
+use Yiisoft\Injector\Injector;
+use function is_array;
+use function is_object;
+use function is_string;
+
+final class ListenerCollectionFactory
+{
+    private Injector $injector;
+    private ContainerInterface $container;
+
+    public function __construct(Injector $injector, ContainerInterface $container)
+    {
+        $this->injector = $injector;
+        $this->container = $container;
+    }
+
+    /**
+     * @param array $eventListeners Event listener list in format ['eventName1' => [$listener1, $listener2, ...]]
+     *
+     * @return ListenerCollection
+     */
+    public function create(array $eventListeners): ListenerCollection
+    {
+        $listenerCollection = new ListenerCollection();
+
+        foreach ($eventListeners as $eventName => $listeners) {
+            if (!is_string($eventName) || !class_exists($eventName)) {
+                throw new InvalidEventConfigurationFormatException(
+                    'Incorrect event listener format. Format with event name must be used.'
```

Comment:
```suggestion
                    "Incorrect event format.  '$eventName' must be event class name."
```

Reply:
Let's not use single quotes in the end-user text. Double quotes only.